### PR TITLE
Make product variants in the lightbox URL accessible.

### PR DESCRIPTION
**Pull request by jkguidaven**
### Description
Supporting URL accessibility for product variants shown in the Lightbox is the primary objective of this PR. I took the opportunity to enhance the way we now activate the product lightbox component. I renamed `useItemLightbox` to `useProductLightbox` in order to better complement the primary component `ProductLightbox` for which it was originally created.

#### Proposed Changes

* [Remove the currentProduct state within the lightbox component and use an external source](https://github.com/Automattic/wp-calypso/commit/273b72d843ce2ba06409b12315ed80168fd44c82). This will make it easier for us to ensure the getters and setters for currentProduct state are coming from the `useProductLightbox` hook. Having the logic for setting the current product within the `useProductLightbox` allows us to hide the implementation of manipulating location hash from consuming components.

* [Rename useItemLightbox to useProductLightbox for better context ](https://github.com/Automattic/wp-calypso/pull/67791/commits/8fb213beeb8fee5d0856349425b97639c921442d) and compliment the `ProductLightbox` component name.

#### Testing Instructions

* 
1. * Run `git fetch && git checkout add/lightbox-url-accessible-product-variant`
    * Run `yarn start-jetpack-cloud`
3. Go to http://jetpack.cloud.localhost:3000/pricing?flags=jetpack/pricing-page-rework-v1 or use the jetpack cloud live link and goto `/pricing?flags=jetpack/pricing-page-rework-v1`.
4. Click on Backup featured or simple card's more info link.
5. Select the 1TB Options
Screen Shot 2022-09-14 at 6 02 10 PM

6. Confirm that `jetpack_backup_t2_yearly` is set in the URL.
Screen Shot 2022-09-14 at 6 03 35 PM

7. Copy/paste the link in a new browser tab.
8. Confirm that the lightbox loads the correct variant during on load.

Screen Shot 2022-09-14 at 6 05 17 PM

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

Related to 1202796695664022-as-1202936358291308

### CODE REVIEW

File: client/my-sites/plans/jetpack-plans/product-lightbox/index.tsx [hunk 1]
```diff
@@ -22,6 +22,7 @@ type Props = ProductStoreBaseProps & {
 	isVisible: boolean;
 	duration: Duration;
 	onClose: () => void;
+	onProductChange: ( product: SelectorProduct | null ) => void;
```
Comment (manzoorwanijk) [inline client/my-sites/plans/jetpack-plans/product-lightbox/index.tsx]:
```suggestion
	onChangeProduct: ( product: SelectorProduct | null ) => void;
```

Comment (jkguidaven) [reply client/my-sites/plans/jetpack-plans/product-lightbox/index.tsx]: Thanks for spotting this.. this is resolved. 👍

**Review by manzoorwanijk — APPROVED** [review]
Works as expected. Thank you keeping it simple

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7527534

Thank you @jkguidaven for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7527534).

**Pull request merged**